### PR TITLE
Fixing 5 bugs

### PR DESCRIPTION
**Pull request by wuda17**

### CODE REVIEW

File: main.c [hunk 1]
```diff
@@ -37,7 +39,7 @@ void blinkGIOB1( void * arg )
     for(;;)
     {
         // Toggle on and off at a set speed
-        gioToggleBit(gioPORTB, 1);
+        gioToggleBit(gioPORTA, 1);
```
Comment (CyStorm) [inline main.c]: @wuda17 what is the rationale behind this change? the LED is mapped to GIO port B pin 1.

Comment (wuda17) [reply main.c]: Oh I thought the first LED was mapped to a different port, I didn't know the second argument represented the pin number

Comment (CyStorm) [reply main.c]: yea, it a little bit unclear of the exact usage of these functions without prior knowledge of the systems, but for the purposes of this where it is just looking for C programming fundamentals, it is fine

**Comment by CyStorm** [comment]
Apart of above comment, everything looks about right. the purpose of this exercise is not for exact correctness against some solution, but for general C programming knowledge.

**Comment by CyStorm** [comment]
looks good
